Yosemite/Yosemite/PointOfSale/PointOfSaleProductService.swift

```diff
@@ -41,13 +40,13 @@ public final class PointOfSaleProductService: PointOfSaleItemServiceProtocol {
     ///
     /// - pageNumber: Number of the page that should be retrieved. If none given, defaults to 1
     ///
-    public func providePointOfSaleItems(pageNumber: Int = 1) async throws -> [POSItem] {
+    public func providePointOfSaleItems(pageNumber: Int = 1) async throws -> (items: [POSItem], hasNextPage: Bool) {
         let productTypes: [ProductType] = isVariableProductsFeatureEnabled ?
         [.simple, .variable] : [.simple]
-        let products = try await productsRemote.loadProductsForPointOfSale(for: siteID, productTypes: productTypes, pageNumber: pageNumber)
+        let (products, totalPagesCount) = try await productsRemote.loadProductsForPointOfSale(for: siteID, productTypes: productTypes, pageNumber: pageNumber)
 
         if pageNumber != 1 && products.count == 0 {
-            throw PointOfSaleProductServiceError.pageOutOfRange
+            return ([], false)
```

Comment:
🗒️ for reviewer: this case should not be reached if the pagination headers are available. When the pagination headers are missing, this "page out of range" case indicates that there there are no more data to load and thus return empty items and `hasNextPage: false`.

Reply:
Thanks. If we want to keep this approach (rather than throwing when the header isn't available/an int) then let's add this comment in the code :)

I think it would also be reasonable to make the headers required though. All tunnelled requests will have them, and I think it's a really old header. We could check whether it's possible to have a site-credential connection without getting these headers back – first step would be to check WC3.5 as I think that's the minimum.